src/cloudai/_core/base_runner.py

```diff
@@ -253,23 +253,24 @@ def get_job_output_path(self, test: Test) -> str:
             test (Test): The test instance for which to generate the output directory path.
 
         Returns:
-            str: The path to the job's output directory.
+            Path: The path to the job's output directory.
 
         Raises:
             ValueError: If the test's section name is None.
             FileNotFoundError: If the base output directory does not exist.
             PermissionError: If there is a permission issue creating the directories.
         """
-        job_output_path = ""
+        if not self.output_path.exists():
+            raise FileNotFoundError(f"Output directory {self.output_path} does not exist")
+
+        job_output_path = None  # Initialize the variable
```

Comment:
Could you please explain why this line is needed?

Reply:
If I don't have this line, I get this error
```
$ ruff format . && ruff check --fix . && pyright .  
129 files left unchanged
All checks passed!
/Users/theo/cloudai-theo/src/cloudai/_core/base_runner.py
  /Users/theo/cloudai-theo/src/cloudai/_core/base_runner.py:273:62 - error: "job_output_path" is possibly unbound (reportPossiblyUnboundVariable)
1 error, 0 warnings, 0 informations 
```

Reply:
Curios that it doesn't report type inconsistency in this case (None vs Path). Let's replace comment this something like `... # avoid reportPossiblyUnboundVariable from pyright` to be explicit about this line.